document-store/src/integrationTest/java/org/hypertrace/core/documentstore/mongo/MongoQueryExecutorTest.java

```diff
@@ -0,0 +1,306 @@
+                        RelationalExpression.of(
+                            IdentifierExpression.of("item"),
+                            IN,
+                            ConstantExpression.ofStrings(List.of("Mirror", "Comb", "Shampoo"))))
+                    .operator(OR)
+                    .operand(
+                        RelationalExpression.of(
+                            IdentifierExpression.of("props.seller.address.pincode"),
+                            EQ,
+                            ConstantExpression.of(700007)))
+                    .build())
+            .addSort(IdentifierExpression.of("props.brand"), ASC)
+            .addSort(IdentifierExpression.of("item"), ASC)
+            .addSort(IdentifierExpression.of("props.seller.address.city"), ASC)
+            .build();
+
+    Iterator<Document> resultDocs = collection.find(query);
+    assertDocsEqual(resultDocs, "mongo/filter_on_nested_fields_response.json");
+  }
+
+  @Test
+  public void testAggregateEmpty() throws IOException {
+    Query query = Query.builder().build();
+
+    Iterator<Document> resultDocs = collection.aggregate(query);
+    assertSizeEqual(resultDocs, "mongo/collection_data.json");
+  }
+
+  @Test
+  public void testAggregateSimple() throws IOException {
+    Query query =
+        Query.builder()
+            .addSelection(AggregateExpression.of(SUM, ConstantExpression.of(1)), "count")
+            .build();
+
+    Iterator<Document> resultDocs = collection.aggregate(query);
+    assertDocsEqual(resultDocs, "mongo/count_response.json");
+  }
+
+  @Test
+  public void testAggregateWithFiltersAndOrdering() throws IOException {
+    Query query =
+        Query.builder()
+            .addSelection(
+                AggregateExpression.of(
+                    SUM,
+                    FunctionExpression.builder()
+                        .operand(IdentifierExpression.of("price"))
+                        .operator(MULTIPLY)
+                        .operand(IdentifierExpression.of("quantity"))
+                        .build()),
+                "total")
+            .addSelection(AggregateExpression.of(FIRST, IdentifierExpression.of("item")), "item")
+            .addAggregation(IdentifierExpression.of("item"))
+            .addSort(IdentifierExpression.of("total"), DESC)
+            .setAggregationFilter(
+                LogicalExpression.builder()
+                    .operand(
+                        RelationalExpression.of(
+                            IdentifierExpression.of("total"), GTE, ConstantExpression.of(25)))
+                    .operator(AND)
+                    .operand(
+                        RelationalExpression.of(
+                            IdentifierExpression.of("total"), LTE, ConstantExpression.of(99)))
+                    .build())
+            .setFilter(
+                RelationalExpression.of(
+                    IdentifierExpression.of("quantity"), NEQ, ConstantExpression.of(10)))
+            .setLimit(10)
```

Comment:
It seems we now two way to set
- `limit` and `offset`
- `pagination`

What is the difference? When should be used explicitly?
If, both are interchangeable, can we just have a single way?

Reply:
Yes. Both are interchangeable. 
"setLimit()" and "setOffset()" are just helper methods to avoid additional (explicit) construction of the Pagination object.

Regarding having a single way, while that's possible,
- "setLimit()" and "setOffset()" are just additional utility methods, which internally build the Pagination object.
- Also, this (having 2 ways) is consistent with all other fields in the Query class. For example, "addSelection()" can be invoked either after constructing the SelectionSpec or by directly invoking with any SelectingExpression. 

In summary, every single member in the Query class has more than one way of defining for the purpose of "syntactic-ease" by avoiding redundant expansions. In other words, "setLimit()" and "setOffset()" are shorthand version of "setPagination()" similar to the other Query-class fields.

Reply:
As discussed, lets keep only as part of follow up PR.